### PR TITLE
feat: remove Keystore

### PR DESCRIPTION
**Pull request by arty-name**
## fixes KILTProtocol/ticket#2040

To avoid over specifying the way to handle private keys, we’d like to replace the usage of Keystore with simple callbacks handling the cryptographic operations. 

## How to test:

yarn test

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/types/src/Keystore.ts [hunk 1]
```diff
@@ -48,98 +49,40 @@ export interface ResponseData<A extends string> {
...
   alg: A
 }
 
 /**
- * A Keystore instance which supports {en/de}cryption using nacl's 'x25519-xsalsa20-poly1305' algorithm.
+ * Methods for {en/de}cryption using nacl's 'x25519-xsalsa20-poly1305' algorithm.
  */
-export interface NaclBoxCapable
-  extends Keystore<any, 'x25519-xsalsa20-poly1305'> {
+export interface NaclBoxCapable {
```
Comment (tjwelde) [inline packages/types/src/Keystore.ts:82]: Should we extract the `encrypt` and `decrypt` properties here into two separate callbacks as we did with the SigningCallback, too?

File: packages/testing/src/TestUtils.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+/**
+ * Generates a keypair usable for signing and a few related values.
+ *
+ * @param alg The algorithm to use for the keypair.
+ * @returns The keypair, matching sign callback, a key usable as DID authentication key.
+ */
+export function makeSigningKeypair(
+  alg: SigningAlgorithms = SigningAlgorithms.Sr25519
+): KeyTool {
```
Comment (tjwelde) [inline packages/testing/src/TestUtils.ts:133]: `makeEncryptionKeypair` returns just the keypair, whereas this function returns KeyTool, which includes more than that. Can we either make the return types consistent (e.g. by returning an encryption keytool from `makeEncryptionKeypair`, which includes the encrypt and decrypt callbacks) or change the naming of this function?

File: packages/core/src/credential/Credential.spec.ts [hunk 3]
```diff
@@ -454,49 +452,40 @@ describe('create presentation', () => {
...
+        ...newKeyForMigratedClaimerDid.authenticationKey,
         id: 'new-auth',
       }
     )
-    const migratedThenDeletedKey = await keystore.generateKeypair({
-      alg: SigningAlgorithms.Ed25519,
-      seed: '//MigratedThenDeletedClaimer',
-    })
+    const migratedThenDeletedKey = makeSigningKeypair(SigningAlgorithms.Ed25519)
+    signMigratedThenDeletedClaimerLightDid = makeSignCallback(
```
Comment (tjwelde) [inline packages/core/src/credential/Credential.spec.ts:484]: can use sign callback from `makeSigningKeypair()`

File: packages/types/src/Keystore.ts (hunk 1 above)
Comment (arty-name) [reply packages/types/src/Keystore.ts:82]: Yes, I was considering this, will do!

File: packages/testing/src/TestUtils.ts (hunk 2 above)
Comment (arty-name) [reply packages/testing/src/TestUtils.ts:133]: I expected this question :) Should they become `makeSigningKeyTool` and `makeEncryptionKeyTool`?

File: tests/bundle-test.ts [hunk 4]
```diff
@@ -41,38 +48,118 @@ function getDefaultMigrationHandler(submitter: KeyringPair) {
   }
 }
 
-async function createFullDidFromSeed(
-  identity: KeyringPair,
-  keystore: InstanceType<typeof Did.DemoKeystore>,
+function makeSignCallback(keypair: KeyringPair): SignCallback {
```
[the remark this answers is not in the document]
Comment (arty-name) [reply tests/bundle-test.ts:51]: Exactly!

**Pull request merged**